### ISSUE TITLE
Add support for autocrypt spec

### ISSUE THREAD

**Issue by ghost** [djcb/mu#1189]
## Expected or desired behavior

Support L1 autocrypt spec by default: https://autocrypt.org/level1.html

## Actual behavior

Manual configuration required for encryption.

## Steps to reproduce

Nothing specific - this is RFE.

## Versions of mu, mu4e/emacs, operating system etc.

Latest.

## Any other detail

The nice thing about autocrypt that it degrades smoothly when communicating with clients which don't support it. It also shouldn't interfere with the existing encryption/signing support - anything explicitly configured by the user should always be preferred. There's interesting challenge as to how to properly implement UI, but once it's there, it should work out of the box.

There're already several implementations available https://autocrypt.org/dev-status.html (including mobiles) so it should be rather straightforward to test.

**Comment by djcb** [comment]
Can you explain in a few lines what this does / why one might want this?

**Comment by osmoc** [comment]
Sure. The idea is rather trivial - make email encryption ubiquitous by making it "transparent" to the users:
- if both sides support it, than it "just works"
- if either side doesn't than it gracefully degrades to regular unprotected communication

More details are in https://autocrypt.org/examples.html

It's not a replacement for proper GPG setup because initial communication is unprotected but it's way easier to adopt for users because it doesn't require any efforts in maintaining web-of-trust etc.

Somewhat lengthier write-up is in https://autocrypt.org/background.html

In short: I'd use GPG to communicate with my nerd friends, I'd use autocrypt to communicate with my girlfriend.

**Comment by ArneBab** [comment]
This is now supported in the newly released Enigmail v2.0 which just hit Ubuntu.

**Comment by ArneBab** [comment]
What this does in practice:

- It adds an [autocrypt header](https://autocrypt.org/level1.html#the-autocrypt-header) which provides the public key to be used for replies: `Autocrypt: addr=a@b.example.org; [prefer-encrypt=mutual;] keydata=BASE64` — this header should be added for outbound mail automatically
- drafts which are synced with the server should be encrypted to the sending account before syncing.
- in can add [gossip headers](https://autocrypt.org/level1.html#key-gossip) to encrypted MIME parts to spread keys within groups
- It defines [setup messages with symmetric encryption](https://autocrypt.org/level1.html#autocrypt-setup-message) which make it easy to set up the same secret keys in multiple email clients, protected by a strong password.

For a practical view, see the [example messages](https://autocrypt.org/level1.html#example-autocrypt-headers).

I hope I got all the essential information. It would be great to have this in mu4e!

**Comment by ArneBab** [comment]
Does this information help? Do you need something more?

**Comment by djcb** [comment]
Ah, thanks. Seems gmime 3.x already has support for this, that helps. It seems a bit non-trivial to add to mu4e though, but I'll keep this in mind.

**Comment by vxcamiloxv** [comment]
Also useful for use mu4e and deltachat at the same time https://delta.chat/en/help#can-i-reuse-my-existing-private-key

**Comment by Grauwolf** [comment]
There seems to be a project adding viewing capabilities to mu4e https://git.sr.ht/~pkal/autocrypt

**Comment by djcb** [comment]
Thanks, @Grauwolf. I'll close the itcket here then.
